### PR TITLE
Upgrading Credit card advanced guide

### PR DESCRIPTION
**Pull request by cherifGsoul**
This upgrades the advanced guide to Canjs V5

### CODE REVIEW

File: docs/can-guides/commitment/recipes/credit-card-advanced/credit-card-advanced.md [hunk 1]
```diff
@@ -30,13 +32,15 @@ To use the widget:
    the invalid inputs should be highlighted red, and the _Pay_
    button should become disabled.
 
-__START THIS TUTORIAL BY CLONING THE FOLLOWING JS BIN__:
+__START THIS TUTORIAL BY CLONING THE FOLLOWING CodePen__:
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/credit-card-advanced/credit-card-advanced.md]: I’d make CodePen all caps too.

File: docs/can-guides/commitment/recipes/credit-card-advanced/1-setup.html [hunk 2]
```diff
@@ -1,20 +1,6 @@
 <!doctype html>
 <title>Credit Card Guide (Advanced)</title>
 <body>
-<script src="https://unpkg.com/kefir@3/dist/kefir.min.js"></script>
-<script src="https://unpkg.com/can@4/dist/global/can.all.js"></script>
 <script src="https://js.stripe.com/v2/"></script>
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/credit-card-advanced/1-setup.html]: It doesn’t look like Stripe is used in this guide. I would remove all these lines.

File: docs/can-guides/commitment/recipes/credit-card-advanced/credit-card-advanced.md [hunk 3]
```diff
@@ -30,13 +32,15 @@ To use the widget:
...
 
-<a class="jsbin-embed" href="https://jsbin.com/woxuhu/2/embed?html,output">
-  Starter Credit Card Guide (Advanced) on jsbin.com
-</a>
+<p class="codepen" data-height="265" data-theme-id="0" data-default-tab="css,result" data-user="cherifGsoul" data-slug-hash="JVPWoZ" style="height: 265px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid black; margin: 1em 0; padding: 1em;" data-pen-title="Credit Card Guide (Starter)">
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/credit-card-advanced/credit-card-advanced.md]: I think it’d be better to start with just the JS and Result tabs if possible.

File: docs/can-guides/commitment/recipes/credit-card-advanced/9-disable-payments.js [hunk 4]
```diff
@@ -1,187 +1,266 @@
...
 			}
-		}
-	});
+		},
+		validateExpiry(expiry) {
+			if (!expiry) {
+				return "There is no expiry. Format  MM-YY";
+			}
+			if (expiry.length !== 2 || expiry[0].length !== 2 || expiry[1].length !== 2) {
+				return "Expirty must be formatted like MM-YY";
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/credit-card-advanced/9-disable-payments.js]: Expirty -> Expiry (here and in other places)

File: docs/can-guides/commitment/recipes/credit-card-advanced/9-disable-payments.js [hunk 5]
```diff
@@ -1,187 +1,266 @@
...
-		]);
-	} else {
-		// STREAM
-		return Kefir.constant({
-			status: "waiting"
-		});
-	}
-});
+		payClicked: {
+			default: 	() => Kefir.emitterProperty()
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/credit-card-advanced/9-disable-payments.js]: Spacing here (and maybe other places?

File: docs/can-guides/commitment/recipes/credit-card-advanced/credit-card-advanced.md (hunk 1 above)
Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/credit-card-advanced/credit-card-advanced.md]: Fixed

File: docs/can-guides/commitment/recipes/credit-card-advanced/1-setup.html (hunk 2 above)
Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/credit-card-advanced/1-setup.html]: Fixed

File: docs/can-guides/commitment/recipes/credit-card-advanced/credit-card-advanced.md (hunk 3 above)
Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/credit-card-advanced/credit-card-advanced.md]: Fixed

File: docs/can-guides/commitment/recipes/credit-card-advanced/9-disable-payments.js (hunk 4 above)
Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/credit-card-advanced/9-disable-payments.js]: Fixed

File: docs/can-guides/commitment/recipes/credit-card-advanced/9-disable-payments.js (hunk 5 above)
Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/credit-card-advanced/9-disable-payments.js]: Fixed

**Pull request merged**